worlds/checksmate/Options.py

```diff
@@ -0,0 +1,415 @@
+
+    No effect on other settings.
+
+    NOT IMPLEMENTED
+    """
+    display_name = "Item Limit By Type"
+
+
+class MaximumItems(ItemDict):
+    """
+    NOT IMPLEMENTED
+
+    For Major To Queen, how many Queen-equivalent pieces you might play with. If set to 1, you will never have more than
+    1 piece upgraded to a Queen. (This does nothing when greater than 'Queen Piece Limit by Type'.) You may still
+    promote pawns during a game. If set to 0, this setting is disabled.
+
+    For Progressive Consul, how many Royal pieces (Kings) to place, which must all be captured before one experiences
+    defeat. The player always starts with 1 King, but may find Progressive Consuls if this is set. Progressive Consuls
+    add additional Kings to the player's starting board.
+
+    For Progressive Pocket, the number of Progressive Pocket Pieces the game is allowed to add to the multiworld. Each
+    Progressive Pocket Piece will improve your 1st, 2nd, or 3rd pocket slot up to 4 times, from Nothing to Pawn, to
+    Minor Piece (like a pocket Knight), to Major Piece (like a pocket Rook), to Queen. This setting does not alter
+    filler item distribution. (Even if you have 0 Progressive Pockets, the item pool may contain Progressive Pocket Gems
+    and Progressive Pocket Range.)
+
+    No effect on other settings.
+
+    NOT IMPLEMENTED
+    """
+    display_name = "Maximum Items"
+
+
+class LockedItems(ItemDict):
+    """
+    Guarantees that these progression and filler items will be unlockable.
+
+    Implementation note: Currently forces this many items into the item pool before distribution begins. This behaviour
+    is not guaranteed - a future version may simply validate the pool contains these items.
+    """
+    display_name = "Locked Items"
+
+
+class DeathLink(Toggle):
+    """
+    If on:
+
+    - Whenever you are checkmated or resign (close the game window), everyone who is also on Death Link dies.
+
+    - Whenever you receive a Death Link event, your game window closes. (You cannot undo or review.)
+    """
+    display_name = "Death Link"
+
+
+cm_options: Dict[str, type(Option)] = {
+    "goal": Goal,
+    "piece_locations": PieceLocations,
+    "piece_types": PieceTypes,
+    "enemy_piece_types": EnemyPieceTypes,
+    "early_material": EarlyMaterial,
+    "min_material": MaterialMinLimit,
+    "max_material": MaterialMaxLimit,
+    "max_engine_penalties": MaximumEnginePenalties,
+    "max_pocket": MaximumPocket,
+    "max_kings": MaximumKings,
+    "fairy_kings": FairyKings,
+    "fairy_chess_pieces": FairyChessPieces,
+    "fairy_chess_army": FairyChessArmy,
+    "fairy_chess_pawns": FairyChessPawns,
+    "minor_piece_limit_by_type": MinorPieceLimitByType,
+    "major_piece_limit_by_type": MajorPieceLimitByType,
+    "queen_piece_limit_by_type": QueenPieceLimitByType,
+    "queen_piece_limit": QueenPieceLimit,
+    "pocket_limit_by_pocket": PocketLimitByPocket,
+    "item_limit_by_type": ItemLimitByType,
```

Comment:
Would comment these out while they're still unfinished, so they don't appear in template yamls.

Reply:
I've commented out just the membership in the dictionary, which I assume is sufficient.

I'm likely to just remove the options, as I think it would be rather confusing to be allowed to put whatever one wants in.

Reply:
These options will live forever in Github. But not in this PR.